test.sh

```diff
@@ -64,10 +97,107 @@ function run_test_coverage() {
   goveralls -v -coverprofile=gover.coverprofile -service=travis-pro
 }
 
+#
+# Main CLI Parser
+#
+USAGE="$(cat -- <<-EOM
+
+Usage:
+Boulder test suite CLI, intended to be run inside of a Docker container:
+
+  docker-compose run --use-aliases boulder ./$(basename "${0}") [OPTION]...
+
+With no options passed: runs standard battery of tests (lint, unit, and integation)
+    -l, --lints                           Adds lint to the list of tests to run
+    -u, --unit                            Adds unit to the list of tests to run
+    -d, --unit-test-filter <DIRECTORY>    Run unit tests for a specific directory
```

Comment:
For long name here here I'd call this `unit-test-packages`, since "filter" suggests to me a value for the `-run` flag, i.e. filtering to specific test cases within a given unittest.

Reply:
How would you feel about `unit-test-dir` instead?

Reply:
I think "packages" is slightly more precise. The `go help test` output describes the relevant argument as "packages." Here's an example of where the semantics are different: `go test sa` fails even though `sa` is a directory, because Go is looking for a package named `sa` in your GOROOT. `go test ./sa` works because in the package naming scheme, that explicitly specifies the current directory. Also `go test ./sa/...` works, even though `./sa/...` isn't a directory.

Reply:
Alright, that makes perfect sense. Thanks.

Reply:
Settled on the singular `--unit-test-package` and `-p` shortopt since the flag can only represent a singular OPTARG but they are parsed in a `while` loop so they both end up in the array.

```
$ docker-compose run --use-aliases boulder ./test.sh -u --unit-test-package=./va --unit-test-package=./ca

Boulder Test Suite CLI

Settings:
    RUN:                unit
    BOULDER_CONFIG_DIR: test/config
    UNIT_FILTER:        ./va ./ca
    RACE:               false
    INT_FILTER:

Starting...

Running Unit Tests
ok  	github.com/letsencrypt/boulder/va	(cached)
ok  	github.com/letsencrypt/boulder/ca	(cached)
SUCCESS
```

Reply:
Excellent. I notice in your example output that the settings are still named `UNIT_FILTER` and `INT_FILTER`, when they should become `FILTER` (based on flag naming feedback elsewhere in this PR) and `UNIT_PACKAGES`.

Reply:
Thanks for looking out, thankfully I did a `Change All Occurrences` for the final renaming